tests/regression/echotest/rands-expected.echo

```diff
@@ -1,8 +1,13 @@
+WARNING: rands() parameter could not be converted in file rands.scad, line 12
```

Comment:
The extra warnings are nice. Is there more info available?
* name or position of the parameter
* target type of the conversation

Reply:
Yes, that information is available. The existing code doesn't have that and I didn't add it, but it could be added easily. No parameter names, because builtin functions mostly do not use named parameters, but parameter position and expected type could be added.

A minor complication is that some functions have multiple possible signatures, which would make the expected parameter type a bit ambiguous. We might need to write some custom warning messages for those cases; for most other cases, the general func.cc check_arguments() function could just add the expected type to the warning message.

Reply:
Ah, good point. As those here are new warnings, we probably should leave it for a separate PR then as it would likely also affect a couple of other places.

I'm thinking if we would have more structured info in code about the parameters, we could generate the documentation instead of the inline text in `Builtins::init()`. That would even bring things one step closer to calltips for user functions.

Reply:
Well too late, I added them :)

Reply:
> No parameter names, because builtin functions mostly do not use named parameters,

Perhaps that should change; perhaps they should always have names for their parameters.

Reply:
While it's possible, it's not really important for things like the math functions. I guess we can have `sin(angle = 30);` but that's not adding much.

Reply:
> Perhaps that should change; perhaps they should always have names for their parameters.

I agree that this would be better. There are a few builtin functions for which this would be tricky, though: min() max() str() chr() are variadic, and log() has a signature of log([base, ] number), and neither of those really fits in the notion of named parameters currently in use -- we'd have to switch to a substantially more elaborate model of parameters to support those.

Reply:
> 
> 
> > No parameter names, because builtin functions mostly do not use named parameters,
> 
> Perhaps that should change; perhaps they should always have names for their parameters.

If you mean mandatory names then -1; I've already had two carpal tunnel operations & I'd prefer to avoid joint replacements.

Reply:
Not mandatory for the user.  Mandatory for the definition of the function, so that the infrastructure can always know the name of a parameter - in the same sense that `cube()` has `size` and `center` parameters even if you invoke it as `cube([1,1,1], true)`.